include/fiction/io/write_operational_domain.hpp

```diff
@@ -121,45 +124,94 @@ inline void write_operational_domain(const operational_domain<parameter_point, o
         throw std::invalid_argument("unsupported number of dimensions in the given operational domain");
     }
 
-    if (num_dimensions == 1)
+    if (!opdom.metric_values.empty())
```

Comment:
If possible, I would always start with a non-negated case in a condition. It makes it easier to read for humans, particularly when it comes to the `else` case where we'd otherwise have to think about a double negation.

Reply:
True! Thanks!